zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -486,65 +523,98 @@ benchmarks! {
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
 
         let categories : u32 = categories.saturated_into();
-        for i in 0..categories.min(d) {
+        for i in 0..d {
             let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+            Pallet::<T>::dispute(
+                RawOrigin::Signed(origin).into(),
+                market_id,
+                OutcomeReport::Categorical((i % 2).saturated_into::<u16>()),
+            )?;
         }
+        let market = T::MarketCommons::market(&market_id)?;
     }: {
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_scalar_reported {
         let total_accounts = 10u32;
         let asset_accounts = 10u32;
-        let (_, market_id) =
-            setup_resolve_common_scalar_after_dispute::<T>(total_accounts, asset_accounts)?;
+        let (caller, market_id) = setup_reported_scalar_market_with_pool::<T>()?;
+        generate_accounts_with_assets::<T>(
+            total_accounts,
+            asset_accounts,
+            Asset::ScalarOutcome(market_id, ScalarPosition::Long),
+        )?;
     }: {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
+        let market = T::MarketCommons::market(&market_id)?;
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_scalar_disputed {
         let total_accounts = 10u32;
         let asset_accounts = 10u32;
         let d in 0..T::MaxDisputes::get();
 
-        let (caller, market_id) =
-            setup_resolve_common_scalar_after_dispute::<T>(total_accounts, asset_accounts)?;
-
+        let (caller, market_id) = setup_reported_scalar_market_with_pool::<T>()?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
+        generate_accounts_with_assets::<T>(
+            total_accounts,
+            asset_accounts,
+            Asset::ScalarOutcome(market_id, ScalarPosition::Long),
+        )?;
```

Comment:
Same question here: What do we need these accounts for?

Reply:
I was under impression of clean_up_pool() to be used from swap() however we have disabled removing account from swaps. So not required.

Reply:
Sorry, I guess the question wasn't totally clear. Even if we didn't disable the removing accounts, why do we need this when we're dealing with scalar markets (where we don't need to destroy accounts)?

Reply:
yes I was not aware about that swaps does not remove accounts for scalar market. It was brought to my attention by @Chralt98 and I have already removed this code. Does it need anything more?

Reply:
No, it's all good. Just wanted to clarify what I meant.